src/node_v8.cc

```diff
@@ -295,6 +300,188 @@ static void SetHeapStatistics(JSONWriter* writer, Isolate* isolate) {
   writer->json_arrayend();
 }
 
+static MaybeLocal<Object> ConvertHeapStatsToJSObject(
+    Isolate* isolate, const cppgc::HeapStatistics& stats) {
+  Local<Context> context = isolate->GetCurrentContext();
+  // Space Statistics
+  LocalVector<Value> space_statistics_array(isolate);
```

Comment:
Minor nit, but if the size is known you can pre-allocate it with the size to speed things up a bit.

```suggestion
  LocalVector<Value> space_statistics_array(isolate, stats.space_stats.size());
```

Reply:
This right way to pre-allocate is to use .reserve() instead. Using the constructor to allocate it like this and then pushing back would lead to the crashes like https://github.com/nodejs/node/issues/56375

Reply:
Ah, good to know. 🙂